frontend/src/scenes/CallingPage.tsx

```diff
@@ -241,10 +256,15 @@ export default function CallingPage({ locale, routePath }: SceneProps) {
     setHasUserDisconnected(true);
   }, [activeConnection]);
 
-  const exitCallingPage = useCallback(() => {
-    disconnectCall();
-    userService?.setShouldSleep(true);
-    contactService?.setActiveContact(null);
+  const handleExitCallButton = useCallback(() => {
+    if (!hasUserDisconnected) {
+      disconnectCall();
```

Comment:
The original callback calls `disconnectCall()` unconditionally. Is there a particular reason why you added this `if`?

Reply:
hmmmm, I can't rmb but relooking at the code yea I think better be safe and call `disconnectCall()` unconditionally.

Reply:
Tested and it's working fine :)